### PR TITLE
pd: add `rank-formula-version` config for hot scheduler

### PR DESCRIPTION
**Pull request by HunDunDM**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add description to the `rank-formula-version` config of PD's balance-hot-region-scheduler. 

Related PD PR: [pd#5515](https://github.com/tikv/pd/pull/5515)

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.4 (TiDB 6.4 versions)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: pd-control.md [hunk 1]
```diff
@@ -1246,6 +1247,16 @@ scheduler config balance-hot-region-scheduler  // 显示 balance-hot-region 调
     scheduler config balance-hot-region-scheduler set enable-for-tiflash true
     ```
 
+- `rank-formula-version` 用于控制热点调度使用的调度策略的算法版本，支持的值有 `["v1", "v2"]`。`v1` 版本为旧有策略，主要关注点为调度是否降低了不同 Store 之间的负载差值，以及是否在另一维度引入副作用。`v2` 版本是新引入的实验性算法，改为关注 Store 之间均衡度的提升率，同时降低了对副作用的关注度，对比 `strict-picking-store` 为 `true` 的 `v1` 算法更注重优先均衡第一维度，对比 `strict-picking-store` 为 `false` 的 `v1` 算法兼顾了第二维度的均衡。在特定场景下，`strict-picking-store` 为 `true` 的 `v1` 版本算法更为保守，只有当两个维度都变好的时候才会进行调度，有可能出现因为维度冲突无法继续均衡的情况，需要将 `strict-picking-store` 改为 `false` 才能取得更好的均衡效果。`v2` 版本算法则可以获得更好的均衡效果并减少无效调度。目前该配置的默认值为 `v1`。 <span class="version-mark">从 v6.3.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline pd-control.md:1250]:
```suggestion
- `rank-formula-version` 用于控制热点调度使用的调度策略的算法版本，支持的值有 `["v1", "v2"]`。目前该配置的默认值为 `v1`。

    - `v1` 版本为 v6.3.0 之前的策略，主要关注点为调度是否降低了不同 Store 之间的负载差值，以及是否在另一维度引入副作用。
    - `v2` 版本是 v6.3.0 引入的实验特性算法，改为关注 Store 之间均衡度的提升率，同时降低了对副作用的关注度。`v2` 版本对比 `strict-picking-store` 为 `true` 的 `v1` 算法更注重优先均衡第一维度，对比 `strict-picking-store` 为 `false` 的 `v1` 算法兼顾了第二维度的均衡。在特定场景下，`strict-picking-store` 为 `true` 的 `v1` 版本算法更为保守，只有当两个维度都变好的时候才会进行调度，有可能出现因为维度冲突无法继续均衡的情况，需要将 `strict-picking-store` 改为 `false` 才能取得更好的均衡效果。`v2` 版本算法则可以获得更好的均衡效果并减少无效调度。
```

File: pd-control.md [hunk 2]
```diff
@@ -1246,6 +1247,16 @@ scheduler config balance-hot-region-scheduler  // 显示 balance-hot-region 调
...
+
+    > **警告：**
+    >
+    > 将 rank-formula-version 配置为 v2 目前为实验性特性，不建议在生产环境中使用。
```
Comment (Oreoxmt) [inline pd-control.md:1254]:
```suggestion
    > 将 `rank-formula-version` 设置为 `v2` 目前为实验性特性，不建议在生产环境中使用。
```

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: pd-control.md [hunk 3]
```diff
@@ -1246,6 +1247,20 @@ scheduler config balance-hot-region-scheduler  // 显示 balance-hot-region 调
     scheduler config balance-hot-region-scheduler set enable-for-tiflash true
     ```
 
+- `rank-formula-version` 用于控制热点调度使用的调度策略的算法版本，支持的值有 `["v1", "v2"]`。目前该配置的默认值为 `v1`。
```
Comment (shichun-0415) [inline pd-control.md:1250]:
```suggestion
- `rank-formula-version` 适用于热点调度，其用来确定调度策略的算法版本，支持的值有 `["v1", "v2"]`。目前该配置的默认值为 `v1`。
```

File: pd-control.md [hunk 4]
```diff
@@ -1246,6 +1247,20 @@ scheduler config balance-hot-region-scheduler  // 显示 balance-hot-region 调
...
+
+    - `v1` 版本为 v6.3.0 之前的策略，主要关注点为调度是否降低了不同 Store 之间的负载差值，以及是否在另一维度引入副作用。
+    - `v2` 版本是 v6.3.0 引入的实验特性算法，改为关注 Store 之间均衡度的提升率，同时降低了对副作用的关注度。`v2` 版本对比 `strict-picking-store` 为 `true` 的 `v1` 算法更注重优先均衡第一维度，对比 `strict-picking-store` 为 `false` 的 `v1` 算法兼顾了第二维度的均衡。
```
Comment (shichun-0415) [inline pd-control.md:1253]:
```suggestion
    - `v2` 版本是 v6.3.0 引入的实验特性算法，主要关注 Store 之间均衡度的提升率，同时降低了对副作用的关注度。对比 `strict-picking-store` 为 `true` 的 `v1` 算法，`v2` 版本更注重优先均衡第一维度，对比 `strict-picking-store` 为 `false` 的 `v1` 算法，`v2` 版本兼顾了第二维度的均衡。
```

File: pd-control.md [hunk 5]
```diff
@@ -1246,6 +1247,20 @@ scheduler config balance-hot-region-scheduler  // 显示 balance-hot-region 调
...
+    - 在特定场景下，`strict-picking-store` 为 `true` 的 `v1` 版本算法更为保守，只有当两个维度都变好的时候才会进行调度，有可能出现因为维度冲突无法继续均衡的情况，需要将 `strict-picking-store` 改为 `false` 才能取得更好的均衡效果。`v2` 版本算法则可以获得更好的均衡效果并减少无效调度。
```
Comment (shichun-0415) [inline pd-control.md:1254]: 两个维度都变好有点模糊，可以换一个清晰点的表述吗？

Comment (shichun-0415) [inline pd-control.md:1254]:
```suggestion
    - 在特定场景下，`strict-picking-store` 为 `true` 的 `v1` 版本算法更为保守，只有当两个维度都变好的时候才会进行调度，有可能出现因为维度冲突无法继续均衡的情况，需要将 `strict-picking-store` 改为 `false` 才能取得更好的均衡效果。`v2` 版本算法则可以获得更好的均衡效果，并减少无效调度。
```

**Comment by shichun-0415** [comment]
@HunDunDM Please help add the experimental feature to the [experimental features](https://docs.pingcap.com/zh/tidb/dev/experimental-features). Thanks.

**Review by lhy1024 — APPROVED** [review]

File: experimental-features.md [hunk 6]
```diff
@@ -26,7 +26,8 @@ aliases: ['/docs-cn/dev/experimental-features-4.0/','/zh/tidb/dev/experimental-f
 
 ## 调度功能
 
-弹性调度功能。结合 Kubernetes，可根据实时负载状态，动态扩缩节点，能够有效地缓解业务高峰的压力并且节约不必要的成本开销。详情参阅：[启用 TidbCluster 弹性伸缩](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/enable-tidb-cluster-auto-scaling)。（v4.0 实验特性）
++ 弹性调度功能。结合 Kubernetes，可根据实时负载状态，动态扩缩节点，能够有效地缓解业务高峰的压力并且节约不必要的成本开销。详情参阅：[启用 TidbCluster 弹性伸缩](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/enable-tidb-cluster-auto-scaling)。（v4.0 实验特性）
++ [调整 PD 中的热点均衡调度器使用的算法版本](./pd-control.md#scheduler-config-balance-hot-region-scheduler)
```
Comment (Oreoxmt) [inline experimental-features.md:30]:
```suggestion
+ 调整 PD 中热点均衡调度器使用 v1 或 v2 版本算法。详情参阅 [`scheduler config balance-hot-region-scheduler`](/pd-control.md#scheduler-config-balance-hot-region-scheduler)。（v6.2 实验特性）
```

File: pd-control.md [hunk 7]
```diff
@@ -1234,12 +1235,26 @@ scheduler config balance-hot-region-scheduler  // 显示 balance-hot-region 调
...
+- `strict-picking-store` 是控制热点调度搜索空间的开关，通常为打开。该配置项仅影响 `rank-formula-version` 为 `v1` 时的行为。当打开时，热点调度的目标是保证所配置的两个维度的热点均衡。当关闭后，热点调度只保证处于第一优先级的维度的热点均衡表现更好，但可能会导致其他维度的热点不再那么均衡。通常用户不需要修改这个配置项。
 
     ```bash
     scheduler config balance-hot-region-scheduler set strict-picking-store true
     ```
 
+- `rank-formula-version` 适用于热点调度，其用来确定调度策略的算法版本，支持的值有 `["v1", "v2"]`。目前该配置的默认值为 `v1`。
+
+    - `v1` 版本为 v6.3.0 之前的策略，主要关注调度是否降低了不同 Store 之间的负载差值，以及是否在另一维度引入副作用。
+    - `v2` 版本是 v6.3.0 引入的实验特性算法，主要关注 Store 之间均衡度的提升率，同时降低了对副作用的关注度。对比 `strict-picking-store` 为 `true` 的 `v1` 算法，`v2` 版本更注重优先均衡第一维度；对比 `strict-picking-store` 为 `false` 的 `v1` 算法，`v2` 版本兼顾了第二维度的均衡。
```
Comment (Oreoxmt) [inline pd-control.md:1247]:
```suggestion
    - `v2` 版本是 v6.3.0 引入的实验特性算法，主要关注 Store 之间均衡度的提升率，同时降低了对副作用的关注度。对比 `strict-picking-store` 为 `true` 的 `v1` 算法，`v2` 版本更注重优先均衡第一维度。对比 `strict-picking-store` 为 `false` 的 `v1` 算法，`v2` 版本兼顾了第二维度的均衡。
```

**Review by Oreoxmt — APPROVED** [review]

File: experimental-features.md [hunk 8]
```diff
@@ -26,7 +26,8 @@ aliases: ['/docs-cn/dev/experimental-features-4.0/','/zh/tidb/dev/experimental-f
 
 ## 调度功能
 
-弹性调度功能。结合 Kubernetes，可根据实时负载状态，动态扩缩节点，能够有效地缓解业务高峰的压力并且节约不必要的成本开销。详情参阅：[启用 TidbCluster 弹性伸缩](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/enable-tidb-cluster-auto-scaling)。（v4.0 实验特性）
++ 弹性调度功能。结合 Kubernetes，可根据实时负载状态，动态扩缩节点，能够有效地缓解业务高峰的压力并且节约不必要的成本开销。详情参阅：[启用 TidbCluster 弹性伸缩](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/enable-tidb-cluster-auto-scaling)。（v4.0 实验特性）
++ 调整 PD 中热点均衡调度器使用 v1 或 v2 版本算法。详情参阅 [`scheduler config balance-hot-region-scheduler`](/pd-control.md#scheduler-config-balance-hot-region-scheduler)。（v6.2 实验特性）
```
Comment (shichun-0415) [inline experimental-features.md:30]: 参考 https://github.com/pingcap/docs-cn/pull/11566/files#diff-8e8a8ffe792c3ade964d617c3e54238e28563d203923a03d29c29bb7f1872360R1247，似乎应该是 v6.3 实验特性？

File: experimental-features.md [hunk 9]
```diff
@@ -26,7 +26,8 @@ aliases: ['/docs-cn/dev/experimental-features-4.0/','/zh/tidb/dev/experimental-f
 
 ## 调度功能
 
-弹性调度功能。结合 Kubernetes，可根据实时负载状态，动态扩缩节点，能够有效地缓解业务高峰的压力并且节约不必要的成本开销。详情参阅：[启用 TidbCluster 弹性伸缩](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/enable-tidb-cluster-auto-scaling)。（v4.0 实验特性）
++ 弹性调度功能。结合 Kubernetes，可根据实时负载状态，动态扩缩节点，能够有效地缓解业务高峰的压力并且节约不必要的成本开销。详情参阅：[启用 TidbCluster 弹性伸缩](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/enable-tidb-cluster-auto-scaling)。（v4.0 实验特性）
```
Comment (shichun-0415) [inline experimental-features.md:29]:
```suggestion
+ 弹性调度功能。结合 Kubernetes，可根据实时负载状态，动态扩缩节点，能够有效地缓解业务高峰的压力并且节约不必要的成本开销。详情参阅[启用 TidbCluster 弹性伸缩](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/enable-tidb-cluster-auto-scaling)。（v4.0 实验特性）
```

File: experimental-features.md (hunk 8 above)
Comment (Oreoxmt) [inline experimental-features.md:30]:
```suggestion
+ 调整 PD 中热点均衡调度器使用 v1 或 v2 版本算法。详情参阅 [`scheduler config balance-hot-region-scheduler`](/pd-control.md#scheduler-config-balance-hot-region-scheduler)。（v6.3 实验特性）
```

**Review by shichun-0415 — APPROVED** [review]

**Pull request merged**